common/utils.go

```diff
@@ -0,0 +1,51 @@
+package common
+
+import (
+	"encoding/hex"
+	"fmt"
+	math "math"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	DustUTXOOffset = 2000
+)
+
+// A very special value to mark current nonce in UTXO
+func NonceMarkAmount(nonce uint64) int64 {
+	if nonce > math.MaxInt64-DustUTXOOffset {
+		panic("nonce mark is too large")
```

Comment:
We should use `panic` in functions in `common`

Reply:
Cool. This panic can impact zetacore. I'll remove this panic and just perform a static cast assuming the nonce will always fall in range of `MaxInt64` which means almost infinite number of transactions.